src/store/fs/util/entity_manager.rs

```diff
@@ -0,0 +1,1322 @@
+                    SelectOutcome::Command(cmd) => {
+                        let Some(cmd) = cmd else {
+                            // Channel closed, this means that the main actor is shutting down.
+                            self.hard_shutdown().await;
+                            break;
+                        };
+                        match cmd {
+                            Command::Spawn(spawn) => {
+                                if let Some(task) = self.state.spawn_boxed(spawn.id, spawn.f).await
+                                {
+                                    self.tasks.spawn(task);
+                                }
+                            }
+                            Command::ShutdownAll(arg) => {
+                                self.soft_shutdown().await;
+                                arg.tx.send(()).ok();
+                                break;
+                            }
+                        }
+                        // Handle incoming command
+                    }
+                    SelectOutcome::Tick(future) => {
+                        if let Some(task) = future {
+                            self.tasks.spawn(task);
+                        }
+                    }
+                    SelectOutcome::TaskDone(result) => {
+                        // Handle completed task
+                        if let Err(e) = result {
+                            error!("Task failed: {e:?}");
+                        }
+                    }
+                }
+            }
+        }
+
+        async fn soft_shutdown(self) {
+            let Self {
+                mut tasks, state, ..
+            } = self;
+            state.shutdown().await;
+            while let Some(res) = tasks.join_next().await {
+                if let Err(e) = res {
+                    eprintln!("Task failed during shutdown: {e:?}");
+                }
+            }
+        }
+
+        async fn hard_shutdown(self) {
+            let Self {
+                mut tasks, state, ..
+            } = self;
+            // this is needed so calls to internal_send in idle shutdown fail fast.
+            // otherwise we would have to drain the channel, but we don't care about the messages at
+            // this point.
+            drop(state);
+            while let Some(res) = tasks.join_next().await {
+                if let Err(e) = res {
+                    eprintln!("Task failed during shutdown: {e:?}");
+                }
+            }
+        }
+    }
+}
+
+/// A manager for entities identified by an entity id.
+///
+/// The manager provides parallelism between entities, but just concurrency within a single entity.
+/// This is useful if the entity wraps an external resource such as a file that does not benefit
+/// from parallelism.
+///
+/// The entity manager internally uses a main actor and per-entity actors. Per entity actors
+/// and their inbox queues are recycled when they become idle, to save allocations.
```

Comment:
Is recycling entity actors in the hot path? Having this optimization costs us a bunch of complexity, another trait (`Reset`), and a bunch of `recycle` fns.
Perhaps the code could be slightly easier but pretty much equally as performant without recycling?